### PR TITLE
[enhancement](merge-on-write) avoid unecessary pk index iteration while memtable flushing

### PR DESCRIPTION
**Pull request by zhannngchen**
## Proposed changes

Issue Number: close #xxx

When load data to new tablet, we don't need  to iterate through the segment primary key index.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buidall

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -2690,11 +2690,16 @@ Status Tablet::calc_delete_bitmap(RowsetSharedPtr rowset,
                                   const RowsetIdUnorderedSet* specified_rowset_ids,
                                   DeleteBitmapPtr delete_bitmap, int64_t end_version,
                                   bool check_pre_segments, RowsetWriter* rowset_writer) {
+    auto rowset_id = rowset->rowset_id();
+    if ((specified_rowset_ids == nullptr || specified_rowset_ids->empty()) && !check_pre_segments) {
```
Comment (liaoxin01) [inline be/src/olap/tablet.cpp:2694]: The rowset with version[0-1] should be considered？

**Comment by zhannngchen** [comment]
run buildall

**Comment by zhannngchen** [comment]
run buildall

**Comment by zhannngchen** [comment]
run buildall

File: be/src/olap/tablet_schema.cpp [hunk 2]
```diff
@@ -492,21 +491,14 @@ bool TabletColumn::is_row_store_column() const {
     return _col_name == BeConsts::ROW_STORE_COL;
 }
```
Comment (liaoxin01) [inline be/src/olap/tablet_schema.cpp:493]: why revert pr #19969？

Comment (zhannngchen) [reply be/src/olap/tablet_schema.cpp]: Thanks for the heads-up. That was my bad. This PR would have caused BE core, so I reverted it during testing. It's all fixed now.

**Comment by zhannngchen** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**